pkg/executor/aggregate/agg_hash_final_worker.go

```diff
@@ -46,11 +47,22 @@ type HashAggFinalWorker struct {
 	partialResultMap    aggfuncs.AggPartialResultMapper
 	BInMap              int
 	isFirstInput        bool
-	groupSet            set.StringSetWithMemoryUsage
 	inputCh             chan *aggfuncs.AggPartialResultMapper
 	outputCh            chan *AfFinalResult
 	finalResultHolderCh chan *chunk.Chunk
 	groupKeys           [][]byte
+
+	spillHelper             *parallelHashAggSpillHelper
+	isSpilledTriggered      bool
+	restoredAggResultMapper *aggfuncs.AggPartialResultMapper
+
+	restoredMemDelta int64
+}
+
+func (w *HashAggFinalWorker) getInputFromDisk() (input *aggfuncs.AggPartialResultMapper, ok bool, spillContinue bool) {
+	ret := w.restoredAggResultMapper
```

Comment:
Will there be some data race since L275 will write to `w.restoredAggResultMapper`

Reply:
> Will there be some data race since L275 will write to `w.restoredAggResultMapper`

No there is no data race, but I think I could merge them together.